helm/hpcc/templates/eclagent.yaml

```diff
@@ -79,7 +80,7 @@ data:
             accessDali: "yes"
             accessEsp: "yes"
         spec:
-          {{- include "hpcc.placementMatchName" (dict "root" $ "pod" (printf "%s-%s-" $apptype .name) "cluster" .name "type" "eclagent") | indent 10 }}
+          {{- include "hpcc.placementsByPodTargetType" (dict "root" $ "pod" $jobNamePrefix "target" .name "type" "eclagent") | indent 10 }}
```

Comment:
so "pod" name becomes the prefix, not the whole pod name (including the ephemeral part) ..
I can see why you've taken that approach (along with comment in values.yaml).
It allows a placement to be defined for e.g. myeclcompiler-compile-

If the pods names (as seen by K8s) don't match this, I think this will be confusing (see other comment, re. changing dynamic pod names too).

But also, it's not obvious that you have to set one of the "pods" list items to a prefix like this.
I think it would be clearer and more flexible if the matching function used a regex, as mentioned in last review, then definition could be e.g.:
```
- pods: ["eclcompilerA.*-compile-.*"]
  placement:
    nodeSelector:
      name: npone
```

or
```
- pods: [".*thor.*-thorworker-.*"]
  placement:
    nodeSelector:
      name: npone
```

Reply:
I did add a function "hpcc.placementsByJobTargetType" passed with "job", "target" and "type". "job" will be regular expression.